main.go

```diff
@@ -194,6 +195,26 @@ func setupReconcilers(mgr ctrl.Manager) {
 		return
 	}
 
+	// Set up a ClusterCacheTracker and ClusterCacheReconciler to provide to controllers
+	// requiring a connection to a remote cluster
+	tracker, err := remote.NewClusterCacheTracker(
+		ctrl.Log.WithName("remote").WithName("ClusterCacheTracker"),
+		mgr,
+	)
+	if err != nil {
+		setupLog.Error(err, "unable to create cluster cache tracker")
+		os.Exit(1)
+	}
+	if _, err := remote.NewClusterCacheReconciler(
+		ctrl.Log.WithName("remote").WithName("ClusterCacheReconciler"),
+		mgr,
+		controller.Options{},
+		tracker,
+	); err != nil {
+		setupLog.Error(err, "unable to create cluster cache reconciler")
+		os.Exit(1)
+	}
```

Comment:
Shouldn't this controller run within a manager?

Reply:
I think that's all handled inside `NewClusterCacheReconciler`

https://github.com/kubernetes-sigs/cluster-api/blob/0e88a403eb18d26301e2693abfc6611e51712502/controllers/remote/cluster_cache.go#L374-L397

Reply:
Looking at https://github.com/kubernetes-sigs/cluster-api/blob/0e88a403eb18d26301e2693abfc6611e51712502/controllers/cluster_controller.go#L74 seems it's not quite enough to actually run the controller.

We should register the reconciler like we do for others in main.go

```
if err := (&controllers.ClusterReconciler{
		Client: mgr.GetClient(),
		Log:    ctrl.Log.WithName("controllers").WithName("Cluster"),
	}).SetupWithManager(mgr, concurrency(clusterConcurrency)); err != nil {
		setupLog.Error(err, "unable to create controller", "controller", "Cluster")
		os.Exit(1)
	}
```

Might be worth maybe to make the ClusterCacheReconciler consistent with the other patterns?

Reply:
I'm not following why this isn't enough, the important part of `SetupWithManager` is the `NewControllerManagedBy()...Build()` chain isn't it?

Which for ClusterCacheReconciler is
https://github.com/kubernetes-sigs/cluster-api/blob/0e88a403eb18d26301e2693abfc6611e51712502/controllers/remote/cluster_cache.go#L387-L390

I'm happy to make it look like the others if you'd prefer. @ncdc Since you did it this way initially, do you remember why, was there a particular benefit?

Reply:
Ah yes you're right, we are indeed registering it here, it'd make sense to be consistent with others

Reply:
I did it this way because I was writing from scratch, and, honestly, I prefer `NewXYZ()` over `&SomeType{...}.SetupWithManager()`. But I realize the rest of the code base is using the latter, and it's the way that kubebuilder generates things.